_posts/2024-02-02-amf-pitch-calibration.md

```diff
@@ -0,0 +1,150 @@
+    const ctxTemplate = templateCanvas.getContext('2d');
+
+    // Set size and color of template canvas
+    ctxTemplate.strokeStyle = 'red';
+    ctxTemplate.fillStyle = "black";
+    ctxTemplate.fillRect(0, 0, templateCanvas.width, templateCanvas.height);
+    
+    // Translate to avoid cropping the sidelines
+    ctxTemplate.lineWidth = lineWidth;
+    ctxTemplate.translate(lineWidth, lineWidth);
+
+    // Side lines
+    addLine({ x: 0, y: 0 }, { x: 0, y: pitchHeight });
+    addLine({ x: pitchWidth, y: 0 }, { x: pitchWidth, y: pitchHeight });
+
+    // End lines
+    addLine({ x: 0, y: 0 }, { x: pitchWidth, y: 0 });
+    addLine({ x: 0, y: pitchHeight }, { x: pitchWidth, y: pitchHeight });
+```
+
+I repeated this process for all pitch elements, resulting in the image below:
+
+<div class="post-center-image">
+    {% picture pimage /assets/images/fullsize/posts/2024-02-02-amf-pitch-calibration/pitchtemplate.png --alt NFL resulting pitch template image %}
+</div>
+
+{:refdef: class="image-caption"}
+*NFL resulting pitch template image*
+{: refdef}
+
+In reality, you see a scaled-up version of a *120 x 53.3 px* image in the web app, adjusted to match your screen size.
+
+### Recovering the homography
+
+The homography is just a mapping that relates the pitch template with the image we upload. This is, that given a point in one of the images, we can find the corresponding point in the other image. There is a lot of cool theory behind this that I do not think I can cover in this post and I also could not improve the work the my colleague Iñaki Rabanillo has done in its own blog. So I will just refer you to his [post](https://iraban.github.io/2021/12/03/homography.html), be sure to check it out since it is a briliant piece of work.
+
+Homography maps the pitch template to the uploaded image, allowing for corresponding points between the two. The theory behind this is extensive and beyond this post's scope. To be fair I can't do better than what my colleague Iñaki Rabanillo has done in its own blog. So I will just refer you to his [post](https://iraban.github.io/2021/12/03/homography.html), be sure to check it out since it is a briliant piece of work.
+
+Summarizing a lot the problem we need to solve, we need to find an homography transformation that is represented by a 3x3 matrix. This will allow us to go from pixel coordinates of real image $p_i$, multiply them by the homography matrix $H$ and obtain the coordinates of point in the template image $p_t$.
+
+$$p_iH=p_t$$
+
+The homography matrix $H$ is a 3x3 matrix that has 8 degrees of freedom. This means that we need at least 4 points in the real image and their corresponding points in the template image to solve for the homography. This is a simple problem to solve and there are a lot of libraries that can do it for you. I used OpenCV library because it is the one I am most familiar with in python and to be honest I do not know if there is any other library that can do it in javascript. 
```

Comment:
```suggestion
The homography matrix $H$ is a 3x3 matrix that has 8 degrees of freedom. This means that we need at least 4 points in the real image and their corresponding pairs in the template image to solve for the homography. This is a simple problem to solve and there are a lot of libraries that can do it for you. I used OpenCV library because it is the one I am most familiar with in python and, to be honest, I am not aware of Javascript specific libraries for this purpose.
```